### PR TITLE
Add wildcard record for Private DNS

### PR DESCRIPTION
**Pull request by day4skiing**
#### What does this PR do?
Double delivery of casl-ansible PR
Adds the DNS wildcard to the private DNS view

#### How should this be manually tested?
From the app-0 VM instance execute: dig @<private IP of dns vm> <env_id.dns_domain> axfr
verify: *.apps.<dns_domain> is pointing to IP of infranode

#### Is there a relevant Issue open for this?
This was delivered to casl-ansible raised issue

#### Who would you like to review this?
cc: @tomassedovic @bogdando PTAL @oybed

### CODE REVIEW

**Comment by day4skiing** [comment]
Note this was tested for functionality in CASL-Ansible, and is a double delivery to openshift-ansible-contrib

File: playbooks/provisioning/openstack/openstack_dns_records.yml [hunk 1]
```diff
@@ -4,6 +4,11 @@
     private_records: "{{ private_records | default([]) + [ { 'type': 'A', 'hostname': hostvars[item]['ansible_hostname'], 'ip': hostvars[item]['private_v4'] } ] }}"
   with_items: "{{ groups['cluster_hosts'] }}"
 
+- name: "Add wildcard records to the private A records for infrahosts"
+  set_fact:
+    private_records: "{{ private_records | default([]) + [ { 'type': 'A', 'hostname': '*.' + openshift_app_domain, 'ip': hostvars[item]['openstack']['private_v4'] } ] }}"
```
Comment (oybed) [inline playbooks/provisioning/openstack/openstack_dns_records.yml]: @day4skiing there's a slight difference between the CASL and the contrib repo - for this PR you need to remove the `[openstack]` portion as the implementation in contrib isn't "openstack aware" at this level.

Comment (day4skiing) [inline playbooks/provisioning/openstack/openstack_dns_records.yml]: Agree... updated pr

**Comment by bogdando** [comment]
with that patch, I can resolve the apps subdomain like `nslookup default.apps.openshift.cluster.lc` et al from a private network, using a private view '*.apps'. LGTM, thank you!

**Pull request merged**